testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -326,19 +327,26 @@ impl EventObserver {
             req.set_body(body);
 
             let response = async_std::task::block_on(async {
-                let stream = match TcpStream::connect(self.endpoint.clone()).await {
-                    Ok(stream) => stream,
-                    Err(err) => {
+                match async_std::future::timeout(
+                    connection_timeout,
```

Comment:
If we're going to change this code path, can we instead please just drop async/await?  It makes using gdb harder than it needs to be, and it pulls in a crap-ton of dependencies

Reply:
No, I'd prefer that this continue to use async/await as it is now.

Reply:
If we decide to do that, it should be in a separate refactoring PR, but I'd prefer not to do that before Nakamoto.